pkg/registry/apis/provisioning/jobs/queue.go

```diff
@@ -125,22 +119,15 @@ func (s *jobStore) drainPending() {
 		}
 
 		var status *provisioning.JobStatus
-		var worker Worker
-		for _, w := range s.workers {
-			if w.Supports(ctx, job) {
-				worker = w
-				break
-			}
-		}
-		if worker == nil {
+		if s.worker == nil {
```

Comment:
I guess was have the worker now and this case doesn't make much sense anymore. 

Or do you want to keep this "registrater" logic for now?